### ISSUE TITLE
Unable to get pointer to Farmer/Node or wrong farmer returned.

### ISSUE THREAD

**Issue by tempestb** [Storj/bridge#324]
### Package Versions

Replace the values below using the output from `npm list storj-bridge`.

```
storj-bridge@0.6.5 /home/gordon/Code/storj-bridge
└── storj@0.6.5  -> /home/gordon/Code/storj-core
```

Replace the values below using the output from `node --version`.

```
v4.2.3
```

### Expected Behavior

Please describe the program's expected behavior. Include an example of your
usage code in the back ticks below if applicable.

```

Call this API request...
 http://api.stoj.io/buckets/57f566390bd56329794d79b8/files/cde36bed6fda14b29e07a7ae?skip=1&limit=1&exclude=""&__nonce=40a154b8-8293-47cb-9455-f12ca0cf18fa

Receive this response...
[{"token":"9e3fcf62be7556326d4ec5de38f5ac1e69b497e9","hash":"dccf1af51df9735f0609fa1a30ecb2357dc3c829","farmer":{"userAgent":"6.0.10","protocol":"1.0.0","address":"client021.storj.dk","port":15039,"nodeID":"d955b53bf3661314c1eccbef9ccd716bd59c153c","lastSeen":1483571534606},"operation":"PULL","size":230672,"index":1}]

```

### Actual Behavior

Please describe the program's actual behavior. Please include any stack traces
or log output in the back ticks below.

```

I instead receive this farmer...
[{"token":"5159d261672c42bfe39d7f5a1384440aa40e6f5f","hash":"dccf1af51df9735f0609fa1a30ecb2357dc3c829","farmer":{"userAgent":"6.0.10","protocol":"1.0.0","address":"104.196.188.40","port":4000,"nodeID":"d955b53bf3661314c1eccbef9ccd716bd59c153c","lastSeen":1483575690286},"operation":"PULL","size":230672,"index":1}]

```

**Please allow me to explain what is going on...**

(Note, in the calls below, the X-Token parameter is setup correctly in the header variables.)

This is being done through the API.  I am not using the CLI or Node.  This is me hitting the server with C# code I've written.

First, I upload a small file.  It only uses 1 shard.  It uploads the file to client021.storj.dk

This works fine.

I then create a PULL token for the file.  This works fine.

I then need to get the pointers to the farmer.  I am unsure if the query string I am creating is correct.  I will explain why in a moment.   The initial call looks like this...

http://api.stoj.io/buckets/57f566390bd56329794d79b8/files/cde36bed6fda14b29e07a7ae?skip=1&limit=1&exclude=""&__nonce=40a154b8-8293-47cb-9455-f12ca0cf18fa

This returns 104.196.188.40 as the farmer.  (Which is wrong)

I send the request to it anyway, and it responds, "Bad Token".   The token isn't bad.  It's the wrong farmer.

Stay with me...

If I use the above query with ANY file hash, I get the SAME FARMER.   104.196.188.40

This may be happening, partially, because the query string is formatted wrong.  I am not sure, because I have nothing to compare it against.  The CLI debug, does not show this command, so I can't compare mine to it.  However, I would expect an error message to be returned, or no response returned like many of the other JSON commands when they fail.  Rather than receiving a bad farmer.  This drove me crazy the last couple of days, because the farmer kept saying, "Bad Token" and I am dumbfounded until I realize it has given me the wrong farmer.  

Further, if I were to change the skip parameter in the query string, to skip=0, (Because really, I'm not skipping any data.)  I don't receive an error.  I just receive an empty JSON return.  []   Which may be the way it handles errors?  I have no idea.   

This makes it very difficult to trouble shoot because...

1.  I have no example to work off of.
2.  When I have it configured one way, it always returns the bad farmer.
3.  When I configure it another way, it returns empty.

I just want a link to the farmer that has the shard my man.  Haha...

Now, in case you think I'm crazy, and I must be sending the wrong hash or something...

I send this (note the file hash in the query string)...
GET
/buckets/57f566390bd56329794d79b8/files/cde36bed6fda14b29e07a7ae
skip=1&limit=2&__nonce=7508efa1-b015-428d-856a-1169a186b153

I receive this...
[{"token":"5159d261672c42bfe39d7f5a1384440aa40e6f5f","hash":"dccf1af51df9735f0609fa1a30ecb2357dc3c829","farmer":{"userAgent":"6.0.10","protocol":"1.0.0","address":"104.196.188.40","port":4000,"nodeID":"d955b53bf3661314c1eccbef9ccd716bd59c153c","lastSeen":1483575690286},"operation":"PULL","size":230672,"index":1}]

I send this (Note the file hash in the query string being different)...
GET
/buckets/57f566390bd56329794d79b8/files/ca7851e29715b4962253b82c
skip=1&limit=2&__nonce=ce4a6eba-4048-443e-9411-653b11a3b22c

I receive this (note this is the same as the response above except the PULL token is different, which would be correct, but the farmer is wrong)...
[{"token":"463e072d00fed779e008edfa97b2d7c3fe085faa","hash":"dccf1af51df9735f0609fa1a30ecb2357dc3c829","farmer":{"userAgent":"6.0.10","protocol":"1.0.0","address":"104.196.188.40","port":4000,"nodeID":"d955b53bf3661314c1eccbef9ccd716bd59c153c","lastSeen":1483581403886},"operation":"PULL","size":230672,"index":1}]

You can see that I've requested two different file ID's from the same bucket.  The result is the exact same hash, farmer, and everything else.  It seems to be defaulting to this node for some reason.  Maybe because the querystring is formatted wrong.  If you could provide an example of how the query string is supposed to look, I will try that instead.  However, I would suggest that it doesn't return a bad farmer when making this call if it is formatted wrong, because other people like me are going to have a difficult time debugging it if we are getting false positives.

I appreciate your help!  I am willing to troubleshoot this with you however I can, just let me know.

Thank You!

**Comment by bookchin** [comment]
When you send the request to the returned farmer, are you including the `x-storj-node-id` header? I noticed that your example for what you expect the farmer to be and what it actually is have the same node ID. Additionally the DK nodes that @th3van is running are strictly set to only tunnel (unless that has changed since the last time we spoke), which means that the farmer the bridge is returning is correct (same expected node ID, but different address) - the farmer has just changed tunnels since you uploaded first. 

When you include the `x-storj-node-id` header (and you always should) while talking to a network node, it's used to determine if the message is for itself or a node it is tunneling. This is probably why you get "bad token" - because that tunneling farmer thinks you are asking it instead of the intended node who is being tunneled.

**Comment by bookchin** [comment]
See https://storj.github.io/core/tutorial-data-transfer.html

**Comment by tempestb** [comment]
I am indeed setting this.  It's virtually the same as uploading.  I am getting the farmer node ID from the result of the query call.  Since it sends the same result (The same wrong farmer) it is going to be the same x-storj-node-id.  That's the problem, the Bridge is sending me the wrong farmer.

**Comment by tempestb** [comment]
It's 2am.  I can't seem to track down what the problem is.  I will resume poking at it in the morning.  I am pretty certain the problem is with my querystring being formatted wrong.  And/Or the ECDSA signed element is wrong.  Somehow this is sending me the wrong farmer.  I will dig more tomorrow and update when I have a better idea of what is going on.  Thanks.

**Comment by tempestb** [comment]
One thing I should point out, is that above I mentioned that the token is not accepted.  That is likely incorrect.  The actual error when I attempt to pick up the shard from the farmer is...

"The remote server returned an error: (401) Unauthorized."

It is likely unauthorized because the farmer is wrong.

Ok...  Example...

I send the request to get the File Pointer with this call...

> http://api.storj.io//buckets/57f566390bd56329794d79b8/files/cde36bed6fda14b29e07a7ae/skip=1&limitmyass=2&__nonce=37daf604-f18c-4e13-a669-64269c922703

Notice that one of the parameters is **LIMITMYASS**

And the result is...

`[{"token":"5a756ebea87139a1414b94237c2f790d9bfd6446","hash":"778bf47ed303fff6d1d01068c578088076b722e4","farmer":{"userAgent":"6.0.11","protocol":"1.0.0","address":"storj13.graymines.net","port":4300,"nodeID":"76cfd5daf1ea80000d158baa9a29fce4a3a472d8","lastSeen":1483632224710},"operation":"PULL","size":293072,"index":1}]`


I tried to elevate the crazy by changing the file's hash to nonsense like "GORDONRULES" but it does throw a 500 error on that one.  So that's good to see.  

I then tried it with NO PARAMETERS except for the nonce.  And it returned the same result.
